src/test/DittoMachine.t.sol

```diff
@@ -646,4 +646,322 @@ contract ContractTest is DSTest, DittoMachine {
         assertEq(ejections, 0);
         assertEq(dm.ownerOf(cloneId), eoa1);
     }
+
+    function testFloorCloneSetsNewClonePrice() public {
+        uint256 nftId = mintNft();
+        address eoa1 = generateAddress("eoa1");
+        address eoa2 = generateAddress("eoa2");
+
+        currency.mint(eoa1, MIN_AMOUNT_FOR_NEW_CLONE * 2);
+        currency.mint(eoa2, MIN_AMOUNT_FOR_NEW_CLONE * 2);
+
+        cheats.startPrank(eoa1);
+        currency.approve(dmAddr, MIN_AMOUNT_FOR_NEW_CLONE * 2);
+
+        uint256 floorId = dm.duplicate(nftAddr, 0, currencyAddr, MIN_AMOUNT_FOR_NEW_CLONE * 2, true);
+        cheats.stopPrank();
+
+        uint256 cloneId = uint256(keccak256(abi.encodePacked(nftAddr, nftId, currencyAddr, false)));
+        assertEq(dm.ownerOf(cloneId), address(0));
+        assertEq(dm.ownerOf(floorId), eoa1);
+
+        cheats.startPrank(eoa2);
+
+        currency.approve(dmAddr, MIN_AMOUNT_FOR_NEW_CLONE * 2);
+
+        cheats.expectRevert(abi.encodeWithSelector(DittoMachine.AmountInvalid.selector));
+        // expect revert with amount less than floor clone worth
+        dm.duplicate(nftAddr, nftId, currencyAddr, MIN_AMOUNT_FOR_NEW_CLONE, false);
+
+        uint256 cId = dm.duplicate(nftAddr, nftId, currencyAddr, MIN_AMOUNT_FOR_NEW_CLONE * 2, false);
+        assertEq(cId, cloneId);
+        cheats.stopPrank();
+
+        uint256 floorMinAmm = dm.getMinAmountForCloneTransfer(floorId);
```

Comment:
Is `Amm` short for amount? If so, would recommend spelling it out to avoid confusion with AMMs.